client/src/pages/Login.tsx

```diff
@@ -1,19 +1,55 @@
 import React from "react";
-import { login } from "../api/auth";
-import { useMutation } from "@tanstack/react-query";
+// import { login } from "../api/auth";
+// import { useMutation } from "@tanstack/react-query";
 import { Link, useNavigate } from "react-router-dom";
+import { useEffect } from "react";
+import { API_URL } from "../api/config";
 
 export const Login = () => {
-  const loginMutation = useMutation({ mutationFn: login });
   const navigate = useNavigate();
 
+  useEffect(() => {
+    console.log("Entering login page");
+    if (localStorage.getItem("token")) {
+      console.log("token exists, navigating to /");
+      navigate("/");
+    }
+  }, []);
+
   const onSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
     event.preventDefault();
     const email = (event.target as HTMLFormElement).email.value;
     const password = (event.target as HTMLFormElement).password.value;
-    console.log(email, password);
-    await loginMutation.mutate({ email, password });
```

Comment:
here as well lets just use react query

Reply:
I tried reverting to that former version, but I think there's some other problem I am not finding... The backend works tho, so it should be a problem in mutation and the exported functions, but have not found it. I like your version better honestly as well, as it simplifies the code, but rn I haven't found a way to debug it. I'll talk to u again in the lab if ur coming!😢